File: src/SuiteCase.Core/Countries/Countries.cs
namespace SuiteCase.Core.Countries;

public static class Countries
{
    public const string DefaultCode = "BG";

    private static readonly Dictionary<string, string> NamesByCode = new(StringComparer.OrdinalIgnoreCase)
    {
        ["AL"] = "Albania",
        ["AT"] = "Austria",
        ["BY"] = "Belarus",
        ["BE"] = "Belgium",
        ["BA"] = "Bosnia and Herzegovina",
        ["BG"] = "Bulgaria",
        ["HR"] = "Croatia",
        ["CY"] = "Cyprus",
        ["CZ"] = "Czechia",
        ["DK"] = "Denmark",
        ["EE"] = "Estonia",
        ["FI"] = "Finland",
        ["FR"] = "France",
        ["DE"] = "Germany",
        ["GR"] = "Greece",
        ["HU"] = "Hungary",
        ["IS"] = "Iceland",
        ["IE"] = "Ireland",
        ["IT"] = "Italy",
        ["LV"] = "Latvia",
        ["LT"] = "Lithuania",
        ["LU"] = "Luxembourg",
        ["MD"] = "Moldova",
        ["ME"] = "Montenegro",
        ["NL"] = "Netherlands",
        ["MK"] = "North Macedonia",
        ["NO"] = "Norway",
        ["PL"] = "Poland",
        ["PT"] = "Portugal",
        ["RO"] = "Romania",
        ["RU"] = "Russia",
        ["RS"] = "Serbia",
        ["SK"] = "Slovakia",
        ["SI"] = "Slovenia",
        ["ES"] = "Spain",
        ["SE"] = "Sweden",
        ["CH"] = "Switzerland",
        ["TR"] = "Turkey",
        ["UA"] = "Ukraine",
        ["GB"] = "United Kingdom"
    };

    public static string NormalizeCodeOrDefault(string? code)
        => string.IsNullOrWhiteSpace(code) ? DefaultCode : code.Trim().ToUpperInvariant();

    public static bool IsSupportedCode(string? code)
        => !string.IsNullOrWhiteSpace(code) && NamesByCode.ContainsKey(code.Trim());

    public static string GetName(string code) => NamesByCode[code.Trim()];
}

File: ui/SuiteCase.Client/src/features/customers/form/countries.ts
export const DEFAULT_COUNTRY_CODE = 'BG';

export const SUPPORTED_COUNTRIES = [
  { code: 'AL', name: 'Albania' },
  { code: 'AT', name: 'Austria' },
  { code: 'BY', name: 'Belarus' },
  { code: 'BE', name: 'Belgium' },
  { code: 'BA', name: 'Bosnia and Herzegovina' },
  { code: 'BG', name: 'Bulgaria' },
  { code: 'HR', name: 'Croatia' },
  { code: 'CY', name: 'Cyprus' },
  { code: 'CZ', name: 'Czechia' },
  { code: 'DK', name: 'Denmark' },
  { code: 'EE', name: 'Estonia' },
  { code: 'FI', name: 'Finland' },
  { code: 'FR', name: 'France' },
  { code: 'DE', name: 'Germany' },
  { code: 'GR', name: 'Greece' },
  { code: 'HU', name: 'Hungary' },
  { code: 'IS', name: 'Iceland' },
  { code: 'IE', name: 'Ireland' },
  { code: 'IT', name: 'Italy' },
  { code: 'LV', name: 'Latvia' },
  { code: 'LT', name: 'Lithuania' },
  { code: 'LU', name: 'Luxembourg' },
  { code: 'MD', name: 'Moldova' },
  { code: 'ME', name: 'Montenegro' },
  { code: 'NL', name: 'Netherlands' },
  { code: 'MK', name: 'North Macedonia' },
  { code: 'NO', name: 'Norway' },
  { code: 'PL', name: 'Poland' },
  { code: 'PT', name: 'Portugal' },
  { code: 'RO', name: 'Romania' },
  { code: 'RU', name: 'Russia' },
  { code: 'RS', name: 'Serbia' },
  { code: 'SK', name: 'Slovakia' },
  { code: 'SI', name: 'Slovenia' },
  { code: 'ES', name: 'Spain' },
  { code: 'SE', name: 'Sweden' },
  { code: 'CH', name: 'Switzerland' },
  { code: 'TR', name: 'Turkey' },
  { code: 'UA', name: 'Ukraine' },
  { code: 'GB', name: 'United Kingdom' },
] as const;

export type SupportedCountryCode = (typeof SUPPORTED_COUNTRIES)[number]['code'];

export type CountryOption = Readonly<{
  code: SupportedCountryCode;
  name: string;
}>;
